apm-lambda-extension/apmproxy/metadata.go

```diff
@@ -39,18 +37,20 @@ func ProcessMetadata(data AgentData) ([]byte, error) {
 	if err != nil {
 		return nil, fmt.Errorf("Error uncompressing agent data for metadata extraction: %w", err)
 	}
-	scanner := bufio.NewScanner(strings.NewReader(string(uncompressedData)))
-	scanner.Scan()
-	if strings.Contains(strings.ToLower(scanner.Text()), "metadata") {
-		return scanner.Bytes(), nil
+
+	before, _, _ := bytes.Cut(uncompressedData, []byte("\n"))
+
+	if bytes.Contains(before, []byte("metadata")) {
+		return before, nil
 	}
```

Comment:
```suggestion
```

To be clear, I don't think we should check it at all, but we should just assume the first line is valid metadata.

Reply:
Ops, understood! I've updated the code 👍